platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/TestingEventBuilder.java

```diff
@@ -0,0 +1,383 @@
+    }
+
+    /**
+     * Set a generation to set for an event. Since the generation is based on the parents generation, this functionality
+     * creates parent descriptors for an event to get the desired generation. It does not work in conjunction with
+     * setting the parents for an event. If parents are set, then the generation will be based on the parents provided.
+     *
+     * @param generation the generation
+     * @return this instance
+     */
+    public @NonNull TestingEventBuilder setGeneration(final Long generation) {
+        fakeGeneration = generation;
+        return this;
+    }
+
+    /**
+     * Set the consensus status of an event. This is only applicable when building an EventImpl.
+     *
+     * @param consensus the consensus status
+     * @return this instance
+     */
+    public @NonNull TestingEventBuilder setConsensus(final boolean consensus) {
+        this.consensus = consensus;
+        return this;
+    }
+
+    private @Nullable EventImpl getSelfParentImpl() {
+        return selfParent instanceof final EventImpl ei ? ei : null;
+    }
+
+    private @Nullable EventImpl getOtherParentImpl() {
+        return otherParent instanceof final EventImpl ei ? ei : null;
+    }
+
+    private @Nullable GossipEvent getSelfParentGossip() {
+        if (selfParent instanceof final GossipEvent ge) {
+            return ge;
+        }
+        return selfParent instanceof final EventImpl ei ? ei.getBaseEvent() : null;
+    }
+
+    private @Nullable GossipEvent getOtherParentGossip() {
+        if (otherParent instanceof final GossipEvent ge) {
+            return ge;
+        }
+        return otherParent instanceof final EventImpl ei ? ei.getBaseEvent() : null;
+    }
+
+    private @NonNull Instant getParentTime() {
+        final Instant sp = getSelfParentGossip() == null
+                ? DEFAULT_TIMESTAMP
+                : getSelfParentGossip().getHashedData().getTimeCreated();
+        final Instant op = getOtherParentGossip() == null
+                ? DEFAULT_TIMESTAMP
+                : getOtherParentGossip().getHashedData().getTimeCreated();
+        return sp.isAfter(op) ? sp : op;
+    }
+
+    /**
+     * Same as {@link #buildGossipEvent()}
+     */
+    public @NonNull GossipEvent buildEvent() {
+        return buildGossipEvent();
+    }
+
+    /**
+     * Build a GossipEvent instance.
+     *
+     * @return the GossipEvent instance
+     */
+    public @NonNull GossipEvent buildGossipEvent() {
+        final ConsensusTransactionImpl[] tr;
+        if (transactions == null) {
+            tr = new ConsensusTransactionImpl[numberOfTransactions];
+            for (int i = 0; i < tr.length; ++i) {
+                final byte[] bytes = new byte[transactionSize];
+                random.nextBytes(bytes);
+                tr[i] = new SwirldTransaction(bytes);
+            }
+        } else {
+            tr = transactions;
+        }
+
+        final long selfParentGen = fakeGeneration != null
```

Comment:
The javadoc for `setGeneration` says "If parents are set, then the generation will be based on the parents provided.", but this implementation looks like a non-null `fakeGeneration` actually is taking precedent over the original generation of a defined parent.